text/0232-maturity-of-otel.md

```diff
@@ -0,0 +1,62 @@
+# Definition of maturity levels to be uniformly used by OpenTelemetry SIGs
+
+On 08 Mar 2023, the OpenTelemetry GC and TC held an OpenTelemetry Leadership summit, discussing various topics. One of the themes we discussed was establishing standard rules for describing the maturity of the OpenTelemetry project. This OTEP summarizes what was discussed there and is intended to have the wider community provide feedback.
+
+This OTEP builds on what was previously communicated by the project, especially the [Versioning and stability for OpenTelemetry clients](https://opentelemetry.io/docs/reference/specification/versioning-and-stability).
+
+The Collector's [stability levels](https://github.com/open-telemetry/opentelemetry-collector#stability-levels) inspired the maturity levels.
+
+## Motivation
+
+Quite often, the community is faced with the question of the quality and maturity expectations of its diverse set of components. This OTEP aims to bring clarity by establishing a framework to communicate the maturity for components delivered by SIGs in the name of the project. As the OpenTelemetry project comprises a multitude of SIGs, and each SIG has several components of varying quality, having this framework will help set the right expectations for OpenTelemetry users using a unified nomenclature.
+
+## Explanation
+
+### Maturity levels
+
+Deliverables of a SIG MUST have a declared maturity level, established by SIG maintainers (SIGs), likely with the input of the code owners. The maturity level of a deliverable is, at most, the lowest level of the user-visible dependencies. For instance, if the Collector's component "otlpreceiver" declares a dependency on the OpenTelemetry Collector API "config" package which is marked with a maturity level of "beta", the "otlpreceiver" can be at most "beta". The OpenTelemetry Collector "core" distribution is also affected so that it can be at most "beta". Only once all dependencies are marked as "stable" MAY a component be marked as "stable".
```

Comment:
>The maturity level of a deliverable is, at most, the lowest level of the user-visible dependencies.

I am not entirely sure about this part. Looks for example at Kubernetes. They have individual features marked [alpha] ([for example](https://kubernetes.io/docs/concepts/containers/images/#maximum-parallel-image-pulls)), however the entire Kubernetes is not marked [alpha], it has a 
stable release 1.x.

I think this is the right approach. Mark individual components appropriately and if the entire thing makes sense for production use then it can still be marked as stable even if there are less mature components in it.

I would prefer to leave it to individual SIGs to decide how they derive the maturity of the whole thing from its constituents.

Reply:
I agree with @tigrannajaryan. I don't like this formulaic approach which would result in every top level components being "beta".

Reply:
There were similar concerns around the Java Agent and I agree with that. I rephrased this.